utils/userData.js

```diff
@@ -0,0 +1,29 @@
+const ORM = require('../');
+const UserData = require('../models/UserData');
+
+/**
+ * a quick way to retreive a user setting stored in user_data
+ * @param {number} userId
+ * @param {string} key
+ * @param {string} _default - fallback value to be used if key not set yet
+ * @returns the stored value
+ */
+module.exports.getUserData = async function(userId, key, _default = undefined) {
+    const row = await UserData.findOne({
+        where: { user_id: userId, key }
+    });
+    return row ? row.data : _default;
+};
+
+/**
+ * a quick way to set or update a user setting in user_data
+ * @param {number} userId
+ * @param {string} key
+ * @param {string} value
+ */
+module.exports.setUserData = async function(userId, key, value) {
+    return ORM.db.query(
+        'INSERT INTO user_data(user_id, `key`, value) VALUES (:userId, :key, :value) ON DUPLICATE KEY UPDATE value = :value, stored_at = CURRENT_TIMESTAMP',
```

Comment:
This looks a lot like `Model.upsert` in Sequelize.
https://sequelize.org/master/class/lib/model.js~Model.html#static-method-upsert

Reply:
yes, as I said in the initial PR comment, I tried using upsert but couldn't get it to work afte 30 minutes so I gave up

Reply:
Sorry, I misread that. Shouldn't work when I still have this headache.